### PR TITLE
Refactor: fix streaming postgresql and redis typing issues

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
A lot of our environment variable handling wasn't safe, and didn't actually work with the types that were expected of `ioredis` and `pg`. This will be updated shortly to move all the redis configuration into a separate `db/redis.js` file and the postgresql code into `db/postgresql.js` files, which further reduces the complexity of `index.js`

### CODE REVIEW

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:144]: This can hopefully all be cleaned up once lands https://github.com/brianc/node-postgres/pull/3128 fixing https://github.com/brianc/node-postgres/issues/2280

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:151]: Previously a `NODE_ENV` of `foo` without a `DATABASE_URL` being set would likely lead to a crash, now we check that the environment is either `"production"` or `"development"` as to assert that we've really a database environment. This would've previously caught us out when starting to build tests, which typically use `NODE_ENV=test`

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:173]: We probably need to handle the potential for a `NaN` case here, so lifting it up something like:

```
let db_pool_max = 10;
if (typeof env.DB_POOL === 'string') {
  const parsedValue = parseInt(env.DB_POOL, 10)
  if (isNaN(parsedValue)) {
    throw new Error(`Invalid DB_POOL environment variable, expected integer, received: ${value}`)
  }
  db_pool_max = parsedValue;
}
```

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:96]: ditto here re handling parseInt & NaN

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:104]: ditto here re handling parseInt & NaN

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:202]: I think this probably also needs to go through a parseInt & NaN check?

File: streaming/index.js
Comment (ThisIsMissEm) [inline streaming/index.js:48]: Needs rebasing once #27828 lands

File: streaming/index.js
Comment (ThisIsMissEm) [reply streaming/index.js]: I've left a FIXME in the code for this.

File: streaming/index.js
Comment (ClearlyClaire) [inline streaming/index.js:163]: I'm not completely sure this is worth echoing back and redacting from. This could marginally be useful, but I guess redacting from could also be marginally confusing.

**Review by ClearlyClaire — DISMISSED** [review]
Looks good overall if a bit verbose. I'd like a second pair of eyes on this though.

File: streaming/index.js
Comment (ThisIsMissEm) [reply streaming/index.js:163]: I could also maybe just drop the database URL entirely from the error throw? I just don't want database credentials ending up in logs, because that'd be really bad.

**Comment by ThisIsMissEm** [comment]
> Looks good overall if a bit verbose. I'd like a second pair of eyes on this though.

it is a bit verbose, but hopefully later we can clean it up a bit once a few upstream dependencies update.

**Review by renchap — DISMISSED** [review]
Looks good to me!

File: streaming/index.js
Comment (ClearlyClaire) [reply streaming/index.js:163]: I think it's best leaving it out, I think a textual replace is probably not the best way to do it since the value could be escaped, thus failing to redact.

Comment (ThisIsMissEm) [reply streaming/index.js:163]:
```suggestion
        throw new Error('Invalid port specified in DATABASE_URL environment variable');
```

**Comment by ThisIsMissEm** [comment]
I can rebase this when I'm home.

**Review by renchap — APPROVED** [review]

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**
